internal/provider/object_user_datasource.go

```diff
@@ -0,0 +1,398 @@
+
+								"secret_key_2": schema.StringAttribute{
+									Computed:            true,
+									Description:         "Second secret key for the object user.",
+									MarkdownDescription: "Second secret key for the object user.",
+								},
+
+								"secret_key_2_exist": schema.BoolAttribute{
+									Computed:            true,
+									Description:         "If the second secret key exists.",
+									MarkdownDescription: "If the second secret key exists.",
+								},
+								"key_timestamp_2": schema.StringAttribute{
+									Computed:            true,
+									Description:         "Timestamp when the second secret key was created.",
+									MarkdownDescription: "Timestamp when the second secret key was created.",
+								},
+								"key_expiry_timestamp_2": schema.StringAttribute{
+									Computed:            true,
+									Description:         "Timestamp when the second secret key expires.",
+									MarkdownDescription: "Timestamp when the second secret key expires.",
+								},
+							},
+						},
+						"tags": schema.ListNestedAttribute{
+							Description:         "Tags associated to the user.",
+							MarkdownDescription: "Tags associated to the user.",
+							Computed:            true,
+							NestedObject: schema.NestedAttributeObject{
+								Attributes: map[string]schema.Attribute{
+									"name": schema.StringAttribute{
+										Description:         "Key of the tag associated to the user.",
+										MarkdownDescription: "Key of the tag associated to the user.",
+										Computed:            true,
+									},
+									"value": schema.StringAttribute{
+										Description:         "Value of the tag associated to the user.",
+										MarkdownDescription: "Value of the tag associated to the user.",
+										Computed:            true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *ObjectUserDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data models.ObjectUser
+	// Load inputs
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var finalUsers []models.ObjectUserDatasourceModel
+	if !data.Name.IsNull() {
+		// CASE 1 — USERNAME PROVIDED
+		username := data.Name.ValueString()
+
+		users, err := d.listUsersByName(ctx, username)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error retrieving object user by name",
+				fmt.Sprintf("Failed retrieving user %s: %v", username, err),
+			)
+			return
+		}
+		finalUsers = users
+	} else if !data.Tag.IsNull() {
+		// CASE 2 — TAG FILTER
+		Namespace := data.Namespace.ValueString()
+		Tag := data.Tag.ValueString()
+		Value := data.Value.ValueString()
+
+		tagUsers, err := d.listUsersByTag(ctx, Namespace, Tag, Value)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				"Error retrieving users by tag",
+				fmt.Sprintf("Unable to retrieve object users with tag filter: %v", err),
```

Comment:
The %v and err formatting does not look good to end users. Please use %s and err.Error().
```suggestion
				fmt.Sprintf("Unable to retrieve object users with tag filter: %s", err.Error()),
```
Or simply just
```suggestion
				err.Error(),
```
should also be good enough.
Applicable to all instances.